modules/metastorage-server/src/main/java/org/apache/ignite/internal/metastorage/server/raft/MetaStorageListener.java

```diff
@@ -230,23 +232,33 @@ else if (clo.command() instanceof RangeCommand) {
                         rangeCmd.keyFrom(),
                         rangeCmd.keyTo());
 
-                cursors.put(cursorId, new IgniteBiTuple<>(cursor, CursorType.RANGE));
+                cursors.put(
+                    cursorId,
+                    new IgniteBiTuple<>(
+                        cursor,
+                        new CursorMeta(CursorType.RANGE, rangeCmd.requesterNodeId())
+                    )
+                );
 
                 clo.result(cursorId);
             }
             else if (clo.command() instanceof CursorNextCommand) {
                 CursorNextCommand cursorNextCmd = (CursorNextCommand) clo.command();
 
-                assert cursors.containsKey(cursorNextCmd.cursorId());
+                IgniteBiTuple<Cursor<?>, CursorMeta> cursorDesc = cursors.get(cursorNextCmd.cursorId());
+
+                if (cursorDesc == null) {
```

Comment:
Why did the cursor can not be found?
I think, need to add an extra information (the node left the cluster), also assertion for that.

Reply:
In real life you should never get here, cause it's only possible to get null if we try to retrieve next element for a node that have already left topology. However it's ease testing dramatically and we still respect the cursor contract.

Reply:
Following comment was added:
```
                // In real life we should never get here, cause it's only possible to get null if we try to retrieve
                // next element for a cursor of a node that has already left topology.
                // However testing with such an approach is much easier and we still respect the cursor's contract.
```